dragonfly/grammar/grammar_base.py

```diff
@@ -450,6 +450,40 @@ def process_begin(self, executable, title, handle):
                               % (self._name,
                                  [r.name for r in self._rules if r.active]))
 
+    def window_change(self, executable, title, handle):
```

Comment:
Why not re-use the `Grammar.process_begin` method?

Reply:
Actually wrote all of this originally over 4 years ago, so I don't remember exactly what I was thinking 😄. But, my guess is that the documentation indicated that `Grammar.process_begin` be called just before the user speaks and I was worried about unintended consequences.

The methods are nearly identical, so reusing should be fine.

Reply:
Haha okay, no problem. The `grammars` property also didn't exist back then.